src/learning/learning.ts

```diff
@@ -90,7 +33,7 @@ function relativeChange (current: TowerOfDerivatives, trial: TowerOfDerivatives)
  *   optimal.   It may be necessary to use additional heuristics (e.g. tabu search) to
  *   find globally optimal assignments.
  */
-export function learnParameters (engine: InferenceEngine, data: PairedObservation[], learningRate: number, maxIterations: number, tolerance: number): LearningResult {
+export function learnParameters (engine: InferenceEngine, data: PairedObservation[], learningRate = 0.1, maxIterations = 100, tolerance = 1E-4): OptimizationResult {
```

Comment:
These are reasonable default values for this function.   I have defined them locally because they do not appear elsewhere.     The constant 1E-4 appears in the Wolfe condition, but is semantically different.